### ISSUE TITLE
bug: ion-card, ion-card-header, ion-card-content, ion-toolbar in conjunction can cause slow rendering

### ISSUE THREAD

**Issue by kevinclarkadstech** [ionic-team/ionic-framework#22097]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

**Expected behavior:**

**Steps to reproduce:**

**Related code:**

```
insert short code snippets here
```

**Other information:**

**Ionic info:**

```
Ionic:

   Ionic CLI       : 6.11.8 (C:\Users\Kevin\AppData\Roaming\npm\node_modules\@ionic\cli)      
   Ionic Framework : @ionic/react 5.3.2        

Capacitor:

   Capacitor CLI   : 2.4.1
   @capacitor/core : 2.4.1

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v12.18.3 (C:\Program Files\nodejs\node.exe)
   npm    : 6.14.6
   OS     : Windows 10
```

I have noticed when rendering a list of ion-cards that it can run slowly starting at 50.  I noticed there is much less lag or delay when using divs to render 50+ cards.  Although there are techniques like virtual scrolling, it can be easier as a developer to append a larger chunk at a time.

I have tried to reproduce things, using two repos, although the delay is not as great as in my actual app.

You can see the "slow" version renders the page header and then takes about 5s to render 200 cards.

The "fast" version renders the page header and all the 200 cards simultaneously.

https://github.com/kevinclarkadstech/ionic-cards-fast

https://github.com/kevinclarkadstech/ionic-cards-slow

**Comment by liamdebeasi** [comment]
I can reproduce this. I do notice about a 1-2 second delay when rendering the "fast" version, but I notice an even longer 5-6 second delay when loading the "slow" version. While the way the page is constructed is maybe not the most efficient, there is definitely something going on with regards to the Ionic/Stencil side that slows things down even more.

I was also able to reproduce this without a framework, and while the delays are not as bad, they are still noticeable. This leads me to believe that an issue on the Ionic/Stencil side is causing React to perform more/expensive re-renders which causes the delay to be even greater.

"Slow" demo: https://codepen.io/liamdebeasi/pen/BaKPeJG
"Fast" demo: https://codepen.io/liamdebeasi/pen/YzqjbNr

I am still investigating this and will let you know when I have more to share. Thanks!

**Comment by liamdebeasi** [comment]
Incidentally, this looks similar to https://github.com/ionic-team/ionic-framework/issues/22095. With the "Slow" demo, Safari renders ~800 layers where as it only renders ~4 layers in the "Fast" demo. That would certainly cause significant slow downs.


@kevinclarkadstech What device/browser did you initially see this issue on?

**Comment by kevinclarkadstech** [comment]
> Incidentally, this looks similar to https://github.com/ionic-team/ionic-framework/issues/22095. With the "Slow" demo, Safari renders ~800 layers where as it only renders ~4 layers in the "Fast" demo. That would certainly cause significant slow downs.
> 
> 
> 
> 
> 
> @kevinclarkadstech What device/browser did you initially see this issue on?

Hmmm good question, but yes I think Safari when I was testing on my MacBook. Usually I test on my Windows laptop in Chrome.

**Comment by kevinclarkadstech** [comment]
Any updates on this?

**Comment by liamdebeasi** [comment]
Does this issue still happen on the latest version of Ionic Framework?

**Comment by kevinclarkadstech** [comment]
I'm not sure because I made a custom container component and am not using these Ionic components anymore.

**Comment by liamdebeasi** [comment]
No problem. We did ship a scroll performance fix in 5.5.0 which should help here: https://github.com/ionic-team/ionic-framework/blob/master/CHANGELOG.md#performance-improvements.

I will close the issue for now, but if you run into scrolling performance problems again feel free to open a new issue. Thanks!
